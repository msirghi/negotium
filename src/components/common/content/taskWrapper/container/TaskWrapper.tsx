import React, { Children, FC } from 'react';
import { PageTitle } from '../../pageTitle/PageTitle';
import { TaskWrapperTitleOptions } from '../../types';
import { Tab, Tabs } from '@mui/material';
import { AddSectionRow } from '../section/add/AddSectionRow';
import { SettingsOptions } from '../../pageTitle/types';
import SmoothList from 'react-smooth-list';
import useTranslation from 'next-translate/useTranslation';
import { DragDropContext, Droppable } from 'react-beautiful-dnd';
import { DndTaskWrapperProps } from '../../../dnd/taskWrapper/types';
import { If } from '../../../utilities/if/If';
import { NoTasks } from '../noTasks/NoTasks';
import { useTheme } from '@mui/styles';

type Props = {
  title: string;
  upperHeaderTitle?: string;
  editableOptions?: TaskWrapperTitleOptions;
  onSectionAdd?: (title: string, orderNumber: number) => void;
  showSections?: boolean;
  taskCount?: number;
} & SettingsOptions &
  Partial<DndTaskWrapperProps>;

export const TaskWrapper: FC<Props> = ({
  title,
  upperHeaderTitle,
  editableOptions,
  children,
  onSectionAdd,
  showSections,
  settingsOptions,
  projectOptions,
  handleDragEnd,
  taskCount,
}) => {
  const { t } = useTranslation('common');
  const theme: any = useTheme();

  const childrenWithProps = React.Children.map(children, (child) => {
    if (!React.isValidElement(child) || !child.props.task) return;
    return React.cloneElement(child, { index: child.props.dndIndex });
  });

  return (
    <div>
      <PageTitle
        editableOptions={editableOptions}
        title={title}
        showUpperHeader
        upperHeaderTitle={upperHeaderTitle}
        settingsOptions={settingsOptions}
        projectOptions={projectOptions}
      />
      <Tabs
        value={1}
        TabIndicatorProps={{
          style: {
            backgroundColor: theme.palette.custom.tabsColor,
          },
        }}
      >
        <Tab
          label={
            <span style={{ color: theme.palette.custom.tabsColor }}>
              {taskCount || ''} {t('common.todos')}`
            </span>
          }
          value={1}
        />
      </Tabs>

      <div role={'tabpanel'} style={{ marginTop: '1rem' }}>
        <If condition={!Children.toArray(children).length}>
          <NoTasks />
        </If>
        <SmoothList>
          <DragDropContext onDragEnd={handleDragEnd!}>
            <Droppable droppableId="list">
              {(provided) => (
                <div ref={provided.innerRef} {...provided.droppableProps}>
                  <>{childrenWithProps}</>
                  {provided.placeholder}
                </div>
              )}
            </Droppable>
          </DragDropContext>
        </SmoothList>
      </div>

      {showSections && <AddSectionRow onSectionSave={(title, orderNumber) => onSectionAdd && onSectionAdd(title, orderNumber)} />}
    </div>
  );
};
