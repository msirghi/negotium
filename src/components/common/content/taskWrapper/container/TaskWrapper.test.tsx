import renderer from 'react-test-renderer';
import { TaskWrapper } from './TaskWrapper';

describe('TaskWrapper', () => {
  const defaultProps = {
    title: 'Title',
  };

  it('should match the snapshot', () => {
    const tree = renderer.create(<TaskWrapper {...defaultProps} />);
    expect(tree).toMatchSnapshot();
  });
});
