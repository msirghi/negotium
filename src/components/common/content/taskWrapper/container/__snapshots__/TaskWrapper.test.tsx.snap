// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskWrapper should match the snapshot 1`] = `
<div>
  <div>
    <div
      onClick={[Function]}
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "width": "initial",
        }
      }
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-c1e8lb-MuiTypography-root"
      >
        Title
      </p>
    </div>
  </div>
  <div
    className="MuiTabs-root css-1ujnqem-MuiTabs-root"
  >
    <div
      className="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
      onScroll={[Function]}
      style={
        Object {
          "marginBottom": -0,
          "overflow": "hidden",
        }
      }
    >
      <div
        aria-orientation={null}
        className="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
        onKeyDown={[Function]}
        role="tablist"
      >
        <button
          aria-selected={true}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={0}
          type="button"
        >
          common.todos
          <span
            className="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
            style={Object {}}
          />
        </button>
        <button
          aria-selected={false}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={-1}
          type="button"
        >
          common.notes
        </button>
      </div>
    </div>
  </div>
  <div
    role="tabpanel"
    style={
      Object {
        "marginTop": "1rem",
      }
    }
  >
    <div />
  </div>
</div>
`;
